lib/Doctrine/Migrations/Metadata/Storage/TableMetadataStorage.php

```diff
@@ -137,4 +137,67 @@ public function complete(ExecutionResult $result) : void
             ]);
         }
     }
+
+    private function checkInitialization() : void
+    {
+        if ($this->runningCheck) {
+            return;
+        }
+        $this->runningCheck      = true;
+        $expectedSchemaChangelog = $this->getExpectedTable();
+
+        if (! $this->isInitialized()) {
+            $this->schemaManager->createTable($expectedSchemaChangelog);
+
+            return;
+        }
+
+        $comparator   = new Comparator();
+        $currentTable = $this->schemaManager->listTableDetails($this->configuration->getTableName());
+        $diff         = $comparator->diffTable($currentTable, $expectedSchemaChangelog);
+        if (! ($diff instanceof TableDiff)) {
+            return;
+        }
+
+        $this->schemaManager->alterTable($diff);
+        $this->updateMigratedVersions();
+    }
+
+    private function isInitialized() : bool
+    {
+        if ($this->connection instanceof MasterSlaveConnection) {
+            $this->connection->connect('master');
+        }
+
+        return $this->schemaManager->tablesExist([$this->configuration->getTableName()]);
+    }
+
+    private function updateMigratedVersions() : void
```

Comment:
This function performs the update of migrated versions from the `YYYYMMDDHHIISS` format to FQCN